### ISSUE TITLE
Feature request: BYOMarkdown parser

### ISSUE THREAD

**Issue by pengwynn** [steveklabnik/metadown#1]
I love Redcarpet, but it would be great to bring your own or outsource Markdown rendering to Tilt.

**Comment by steveklabnik** [comment]
Tilt is a Good Idea. +1 for sure.

**Comment by postmodern** [comment]
+1 Less coupling for great justice.

**Comment by steveklabnik** [comment]
It appears like Tilt may sacrifice the ability to render Strings. Not 100% sure how I feel about that.

**Comment by postmodern** [comment]
I swear there is a multi-markdown gem, or am I imagining things?

**Comment by steveklabnik** [comment]
I was thinking it'd work for any kind of Tilt-able template.

**Comment by steveklabnik** [comment]
Do a `gem install metadown --pre`.

84f4bb751cf7fd3 is the commit. Basically, you can inject your own Tilt-compatible parser.

What do you guys think?

**Comment by pengwynn** [comment]
Awesome! Love it. Now I just need to work on a patch for Nesta...

**Comment by steveklabnik** [comment]
Awesome.

It'd be cool if I could actually get this into Tilt proper, or make it some kind of extension, but this will do for now...

**Comment by miloshadzic** [comment]
I don't think that this is a good idea. At least it kinda sucks for my use case:

I'd like a somewhat custom Markdown renderer for example something like:

``` ruby
class PygmentsMarkdown < Metadown::Renderer
  include Redcarpet::Render::SmartyPants

  def block_code(code, language)
    Pygments.highlight(code, :lexer => language)
  end
end
```

so that I can have a different renderer but still get the YAML metadata from metadown. In the current version I can't do that AFAIK unless I create a Redcarpet renderer and adapt it to the Tilt interface.

[I forked Metadown](https://github.com/miloshadzic/metadown) and [changed the code and specs](https://github.com/miloshadzic/metadown/commit/1dfa9987fe226fb169ac67ce99d3e88000ea8ffa) (didn't touch the README though) and now I can use it like this:

``` ruby
class Post
  def initialize(source)
    @post_data = Metadown.render(source,
                     Redcarpet::Markdown.new(PygmentsMarkdown, :fenced_code_blocks => true))
  end
  ...
end
```

Let me know what you think.

**Comment by steveklabnik** [comment]
Hmm. Okay. Seems legit. Can you send me a pull request?

**Comment by miloshadzic** [comment]
Sure, just let me update the README. Wanted to see what you think first.

**Comment by steveklabnik** [comment]
No worries. :) Take your time.
